examples/ner/run_ner.py

```diff
@@ -596,26 +599,69 @@ def main():
 
     args.model_type = args.model_type.lower()
     config_class, model_class, tokenizer_class = MODEL_CLASSES[args.model_type]
-    config = config_class.from_pretrained(
-        args.config_name if args.config_name else args.model_name_or_path,
-        num_labels=num_labels,
-        id2label={str(i): label for i, label in enumerate(labels)},
-        label2id={label: i for i, label in enumerate(labels)},
-        cache_dir=args.cache_dir if args.cache_dir else None,
-    )
     tokenizer_args = {k: v for k, v in vars(args).items() if v is not None and k in TOKENIZER_ARGS}
     logger.info("Tokenizer arguments: %s", tokenizer_args)
     tokenizer = tokenizer_class.from_pretrained(
         args.tokenizer_name if args.tokenizer_name else args.model_name_or_path,
         cache_dir=args.cache_dir if args.cache_dir else None,
         **tokenizer_args,
     )
-    model = model_class.from_pretrained(
-        args.model_name_or_path,
-        from_tf=bool(".ckpt" in args.model_name_or_path),
-        config=config,
-        cache_dir=args.cache_dir if args.cache_dir else None,
-    )
+
+    if args.has_new_labels and os.path.exists(args.model_name_or_path):
+        old_config = config_class.from_pretrained(
+            args.config_name if args.config_name else args.model_name_or_path,
+            cache_dir=args.cache_dir if args.cache_dir else None,
+        )
+        old_model = model_class.from_pretrained(
+            args.model_name_or_path,
+            from_tf=bool(".ckpt" in args.model_name_or_path),
+            config=old_config,
+            cache_dir=args.cache_dir if args.cache_dir else None,
+        )
+        old_num_labels = old_config.num_labels
+        new_labels = [label for label in labels if label not in old_config.label2id.keys()]
+        id2label = config.id2label + {str(i+old_num_labels): label for i, label in enumerate(new_labels)}
+        label2id = config.id2label + {str(i+old_num_labels): label for i, label in enumerate(new_labels)}
```

Comment:
Are these two lines identical save the name?
```suggestion
        id2label = config.id2label + {str(i+old_num_labels): label for i, label in enumerate(new_labels)}
        label2id = id2label
```

Reply:
I commented on this before you changed it :sweat_smile: